source/io/module_eggShell.js

```diff
@@ -239,6 +246,257 @@
+            return Module._TypeRef_IsBoolean(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsInteger = function (typePointer) {
+            return Module._TypeRef_IsInteger(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsSigned = function (typePointer) {
+            return Module._TypeRef_IsSigned(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsEnum = function (typePointer) {
+            return Module._TypeRef_IsEnum(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsFloat = function (typePointer) {
+            return Module._TypeRef_IsFloat(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsString = function (typePointer) {
+            return Module._TypeRef_IsString(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsPath = function (typePointer) {
+            return Module._TypeRef_IsPath(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsTimestamp = function (typePointer) {
+            return Module._TypeRef_IsTimestamp(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsComplex = function (typePointer) {
+            return Module._TypeRef_IsComplex(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsAnalogWaveform = function (typePointer) {
+            return Module._TypeRef_IsAnalogWaveform(typePointer) !== 0;
+        };
+
+        var validateVisitMethod = function (fn, fnName) {
+            if (typeof fn !== 'function') {
+                throw new Error('Visitor must have a method named `' + fnName + '`. Found: ' + fn);
+            }
+        };
+
+        var dispatchVisitBoolean = function (typeVisitor, args) {
+            validateVisitMethod(typeVisitor.visitBoolean, 'visitBoolean');
+            return typeVisitor.visitBoolean.apply(typeVisitor, args);
+        };
+
+        var dispatchVisitEnum = function (typeVisitor, args) {
+            validateVisitMethod(typeVisitor.visitEnum, 'visitEnum');
+            return typeVisitor.visitEnum.apply(typeVisitor, args);
+        };
+
+        var dispatchVisitInteger = function (typeVisitor, args) {
+            var typeRef = args[0].typeRef;
+            var isSignedInteger = Module.eggShell.typeIsSigned(typeRef);
+            var sizeOfInteger = Module.eggShell.typeTopAQSize(typeRef);
+            var visitFn = undefined;
+            if (isSignedInteger === true) {
+                switch (sizeOfInteger) {
+                case 1:
+                    visitFn = typeVisitor.visitInt8;
+                    break;
+                case 2:
+                    visitFn = typeVisitor.visitInt16;
+                    break;
+                case 4:
+                    visitFn = typeVisitor.visitInt32;
+                    break;
+                case 8:
+                    visitFn = typeVisitor.visitInt64;
+                    break;
+                default:
+                    throw new Error('Unexpected size for Integer');
```

Comment:
Add more debugging information to the error as follows: "Unexpected integer size. Found: " + sizeOfInteger. That way if something bad happens you won't wonder about the integer size that caused the exception.